airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -166,6 +166,17 @@ private static DSLContext getDslContext(final JsonNode config) {
         SQLDialect.POSTGRES);
   }
 
+  private void revokeReplicationPermission() {
+    executeQuery("ALTER USER " + container.getUsername() + " NOREPLICATION;");
+  }
+
+  @Test
+  void testCheckWithoutPrivileges() throws Exception {
+    revokeReplicationPermission();
+    final AirbyteConnectionStatus status = source.check(config);
+    assertEquals(status.getStatus(), AirbyteConnectionStatus.Status.FAILED);
```

Comment:
super small nit: Can you also test the exact error message here as well?

Reply:
Done ✅